activesupport/lib/active_support/cache.rb

```diff
@@ -12,6 +12,15 @@
 module ActiveSupport
   # See ActiveSupport::Cache::Store for documentation.
   module Cache
+    class InvalidCacheKeyError < StandardError
+      def initialize(key)
+        super("You are attempting to use a key: `#{key.inspect}` which is not\n" \
+          "supported because it is blank. If you are intentionally" \
+          "trying to use the cache with a blank key, please instead pass in\n" \
+          "a string such as: `'intentionally_empty'`")
+      end
+    end
```

Comment:
Think we can safely steer folks away from blank/missing keys and drop the workaround suggestion? Those with a specific need can dig deeper to understand it and change their approach.

Then a specific exception with little or no message could be sufficient, e.g. `BlankCacheKeyError` or `MissingKeyError` saying "Cache keys may not be nil or empty: #{key.inspect}"